File: app/src/main/java/net/squanchy/navigation/deeplink/DeepLinkParsingException.java
package net.squanchy.navigation.deeplink;

final class DeepLinkParsingException extends RuntimeException {

    static DeepLinkParsingException eventDetailsMissingEventId(String path) {
        return new DeepLinkParsingException("Event details deeplink is missing event ID: \"" + path + "\"");
    }

    static DeepLinkParsingException speakerDetailsMissingSpeakerId(String path) {
        return new DeepLinkParsingException("Speaker details deeplink is missing speaker ID: \"" + path + "\"");
    }

    private DeepLinkParsingException(String message) {
        super(message);
    }
}

File: app/src/main/java/net/squanchy/navigation/deeplink/DeepLinkParsingException.kt
package net.squanchy.navigation.deeplink

internal class DeepLinkParsingException private constructor(message: String) : RuntimeException(message) {

    companion object {

        fun eventDetailsMissingEventId(path: String): DeepLinkParsingException {
            return DeepLinkParsingException("Event details deeplink is missing event ID: \"$path\"")
        }

        fun speakerDetailsMissingSpeakerId(path: String): DeepLinkParsingException {
            return DeepLinkParsingException("Speaker details deeplink is missing speaker ID: \"$path\"")
        }
    }
}
